containers/daap-delete-table-for-data-product/src/var/task/delete_table.py

```diff
@@ -0,0 +1,123 @@
+        print(f"deleted {number_of_files} data files from {prefix}")
+
+
+def handler(event, context):
+    """
+    Handles requests that are passed through the Amazon API Gateway data-product/ REST API endpoint.
+    POST requests result in a success code to report that the product has been registered
+    (metadata has been created).
+    GET, PUT, and DELETE requests result in a 405 response.
+
+    :param event: The event dict sent by Amazon API Gateway that contains all of the
+                  request data.
+    :param context: The context in which the function is called.
+    :return: A response that is sent to Amazon API Gateway, to be wrapped into
+             an HTTP response. The 'statusCode' field is the HTTP status code
+             and the 'body' field is the body of the response.
+    """
+
+    data_product_name = event["pathParameters"].get("data-product-name")
+    table_name = event["pathParameters"].get("table-name")
+
+    logger.add_extras(
+        {"data_product_name": data_product_name, "table_name": table_name}
+    )
+    logger.info(f"event: {event}")
+
+    data_product_metadata = DataProductMetadata(
+        data_product_name=data_product_name, logger=logger, input_data=None
+    )
+    if not data_product_metadata.exists:
+        error_message = (
+            f"Could not locate metadata for data product: {data_product_name}."
+        )
+        logger.error(error_message)
+        return format_error_response(HTTPStatus.BAD_REQUEST, event, error_message)
+
+    table_schema = DataProductSchema(
+        data_product_name=data_product_name,
+        table_name=table_name,
+        logger=logger,
+        input_data=None,
+    )
+
+    # Validate the existence of the table schema for the latest data product version
+    if not table_schema.exists:
+        error_message = f"Could not locate valid schema for table: {table_name}."
+        logger.error(error_message)
+        return format_error_response(HTTPStatus.BAD_REQUEST, event, error_message)
+
+    # Attempt deletion of the glue table
+    try:
+        glue_client.get_table(DatabaseName=data_product_name, Name=table_name)
+    except ClientError as e:
+        if e.response["Error"]["Code"] == "EntityNotFoundException":
+            error_message = f"Could not locate glue table '{table_name}' in database '{data_product_name}'"
+            logger.error(error_message)
+        else:
+            error_message = f"Unexpected ClientError: {e.response['Error']['Code']}"
+            logger.error(error_message)
+        return format_error_response(HTTPStatus.BAD_REQUEST, event, error_message)
+    else:
+        glue_client.delete_table(DatabaseName=data_product_name, Name=table_name)
+
+    # Proceed to delete the raw data
+    element = DataProductElement.load(
+        element_name=table_name, data_product_name=data_product_name
+    )
+
+    raw_bucket = element.data_product.raw_data_bucket
```

Comment:
I think this will be scoped to the current version only, so we won't delete any historical data (which I think we wanted to do)

I'm wondering if we need to add some way to get all the paths to historical versions of a table, since they won't have a common prefix.

`/raw/some-product/v1.2/some-table/...` vs `/raw/some-product/v1.1/some-table/...`

Reply:
Yes agree, thought that there was agreement just to delete current data but actually not sure that this makes sense any longer.